tests/Api/JsonApiTestCase.php

```diff
@@ -74,7 +74,7 @@ protected function assertResponseViolations(Response $response, array $expectedV
     protected function assertJsonResponseViolations(Response $response, array $expectedViolations): void
     {
         $violations = json_decode($response->getContent(), true)['violations'] ?? [];
```

Comment:
```suggestion
        $responseContent = $response->getContent() ?: '';
        $this->assertNotEmpty($responseContent);

        $violations = json_decode($responseContent, true)['violations'] ?? [];
```
Failing early and not re-accessing the same thing multiple times